include/seqan3/alphabet/union_alphabet.hpp

```diff
@@ -258,11 +308,36 @@ struct union_alphabet
         return value >= rhs.value;
     }
     //!@}
+
+protected:
+    //! \privatesection
+    // conversion tables
+
+    constexpr auto from_base_(const variant_type & alphabet_v) const {
+        return std::visit([&](auto && alphabet)
+        {
+            return prefix_sum_sizes[alphabet_v.index()] + alphabet.to_integral();
+        }, alphabet_v);
+    }
+
+    static constexpr auto prefix_sum_sizes
+        = detail::alphabet_prefix_sum_sizes<first_alphabet_type, alphabet_types...>();
+
+    static constexpr auto value_to_char
+        = detail::union_alphabet::value_to_char_table<char_type, first_alphabet_type, alphabet_types...>();
+
+    static constexpr auto char_to_value
+        = detail::union_alphabet::char_to_value_table<char_type, first_alphabet_type, alphabet_types...>();
 };
 
+} // namespace seqan3
+
 #ifndef NDEBUG
-static_assert(alphabet_concept<union_alphabet<dna5, dna5>>);
-static_assert(detail::internal_alphabet_concept<union_alphabet<dna5, dna5>>);
+#include "nucleotide/dna5.hpp"
+static_assert(std::is_pod_v<seqan3::union_alphabet<seqan3::dna5, seqan3::dna5>>);
+static_assert(std::is_trivial_v<seqan3::union_alphabet<seqan3::dna5, seqan3::dna5>>);
+static_assert(std::is_trivially_copyable_v<seqan3::union_alphabet<seqan3::dna5, seqan3::dna5>>);
+static_assert(std::is_standard_layout_v<seqan3::union_alphabet<seqan3::dna5, seqan3::dna5>>);
+static_assert(seqan3::alphabet_concept<seqan3::union_alphabet<seqan3::dna5, seqan3::dna5>>);
```

Comment:
It didn't hurt to have better diagnostics, what broke. Since it is not clear, that this class is pod and conforms to the standard layout etc... I would leave it.